ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/LoadingPaywallView.kt

```diff
@@ -0,0 +1,148 @@
+package com.revenuecat.purchases.ui.revenuecatui
+
+import android.content.Context
+import androidx.compose.foundation.layout.Box
+import androidx.compose.material3.ColorScheme
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.platform.LocalContext
+import androidx.lifecycle.ViewModel
+import com.revenuecat.purchases.Offering
+import com.revenuecat.purchases.Package
+import com.revenuecat.purchases.PackageType
+import com.revenuecat.purchases.models.Period
+import com.revenuecat.purchases.models.Price
+import com.revenuecat.purchases.paywalls.PaywallData
+import com.revenuecat.purchases.ui.revenuecatui.data.PaywallViewModel
+import com.revenuecat.purchases.ui.revenuecatui.data.PaywallViewState
+import com.revenuecat.purchases.ui.revenuecatui.data.processed.PaywallTemplate
+import com.revenuecat.purchases.ui.revenuecatui.data.processed.TemplateConfiguration
+import com.revenuecat.purchases.ui.revenuecatui.data.processed.VariableDataProvider
+import com.revenuecat.purchases.ui.revenuecatui.data.testdata.MockApplicationContext
+import com.revenuecat.purchases.ui.revenuecatui.data.testdata.TestStoreProduct
+import com.revenuecat.purchases.ui.revenuecatui.extensions.createDefault
+import com.revenuecat.purchases.ui.revenuecatui.helpers.toAndroidContext
+import com.revenuecat.purchases.ui.revenuecatui.helpers.toPaywallViewState
+import com.revenuecat.purchases.ui.revenuecatui.templates.Template2
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import java.net.URL
+
+@Composable
+internal fun LoadingPaywallView(mode: PaywallViewMode) {
+    val paywallData: PaywallData = PaywallData.createDefault(
+        LoadingPaywallConstants.packages,
+        MaterialTheme.colorScheme,
+    )
+
+    val offering = Offering(
+        identifier = LoadingPaywallConstants.offeringIdentifier,
+        serverDescription = "Loading paywall",
+        metadata = emptyMap(),
+        availablePackages = LoadingPaywallConstants.packages,
+        paywall = paywallData,
+    )
+
+    val viewModel = LoadingViewModel(mode, offering, paywallData)
```

Comment:
Classes that inherit from `ViewModel` should not be instantiated using the constructor. Instead you could use the `viewModel` function to get an instance. However, you would need to create a factory to pass this data to the VM, similar to how we create the main view model currently

Alternatively, we could just make this VM not inherit from `ViewModel`. The main advantage of using that is that you get a data container that survives configuration changes automatically... But I don't think that's that important for this. If we see performance issues at some point, we can reconsider.

Reply:
TIL, thanks! I removed the `ViewModel` parent class.